test/Benchmarks/Logs/LogBenchmarks.cs

```diff
@@ -7,23 +7,24 @@
 using OpenTelemetry.Logs;
 
 /*
-BenchmarkDotNet v0.13.10, Windows 11 (10.0.22621.3007/22H2/2022Update/SunValley2)
+BenchmarkDotNet v0.13.10, Windows 11 (10.0.22631.3880/23H2/2023Update/SunValley3)
 11th Gen Intel Core i7-1185G7 3.00GHz, 1 CPU, 8 logical and 4 physical cores
-.NET SDK 8.0.101
-  [Host]     : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2
-  DefaultJob : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2
-
-
-| Method                        | Mean       | Error     | StdDev    | Gen0   | Allocated |
-|------------------------------ |-----------:|----------:|----------:|-------:|----------:|
-| NoListenerStringInterpolation | 124.458 ns | 2.5188 ns | 2.2329 ns | 0.0114 |      72 B |
-| NoListenerExtensionMethod     |  36.326 ns | 0.2916 ns | 0.2435 ns | 0.0102 |      64 B |
-| NoListener                    |   1.375 ns | 0.0586 ns | 0.0896 ns |      - |         - |
-| UnnecessaryIsEnabledCheck     |   1.332 ns | 0.0225 ns | 0.0188 ns |      - |         - |
-| CreateLoggerRepeatedly        |  48.295 ns | 0.5951 ns | 0.4970 ns | 0.0038 |      24 B |
-| OneProcessor                  |  98.133 ns | 1.8805 ns | 1.5703 ns | 0.0063 |      40 B |
-| TwoProcessors                 | 105.414 ns | 0.4610 ns | 0.3850 ns | 0.0063 |      40 B |
-| ThreeProcessors               | 102.023 ns | 1.4187 ns | 1.1847 ns | 0.0063 |      40 B |
+.NET SDK 8.0.107
+  [Host]     : .NET 8.0.7 (8.0.724.31311), X64 RyuJIT AVX2
+  DefaultJob : .NET 8.0.7 (8.0.724.31311), X64 RyuJIT AVX2
+
+
+| Method                        | Mean       | Error     | StdDev     | Median     | Gen0   | Gen1   | Allocated |
+|------------------------------ |-----------:|----------:|-----------:|-----------:|-------:|-------:|----------:|
+| NoListenerStringInterpolation | 135.503 ns | 2.7458 ns |  4.5114 ns | 135.391 ns | 0.0114 |      - |      72 B |
+| NoListenerExtensionMethod     |  40.218 ns | 0.8249 ns |  2.2581 ns |  39.809 ns | 0.0102 |      - |      64 B |
+| NoListener                    |   1.930 ns | 0.0626 ns |  0.1264 ns |   1.889 ns |      - |      - |         - |
+| UnnecessaryIsEnabledCheck     |   1.531 ns | 0.0542 ns |  0.1267 ns |   1.518 ns |      - |      - |         - |
+| CreateLoggerRepeatedly        |  53.797 ns | 1.0927 ns |  1.7331 ns |  53.401 ns | 0.0038 |      - |      24 B |
+| OneProcessor                  | 111.558 ns | 2.9821 ns |  8.5082 ns | 109.311 ns | 0.0063 |      - |      40 B |
+| BatchProcessor                | 263.650 ns | 5.2908 ns | 14.1223 ns | 258.984 ns | 0.0200 | 0.0043 |     128 B |
```

Comment:
@CodeBlanch What might be the reason why there is a notable difference when the `LogRecord` comes from a heap pool vs thread-local? There is no extra copy involved...

Reply:
Do we just want to measure the perf difference specific to the pooling mechanism? If yes, then the benchmarks should just compare `LogRecordThreadStaticPool.Instance.Rent()`  vs `LogRecordSharedPool.Current.Rent()`.

Testing this using `BatchProcessor` would add all the `BatchProcessor` related overhead as well such as buffering the log record attributes, adding data to `CircularBuffer` etc.

Reply:
I want to measure the overall cost when using Batch vs Reentrant processors.

Anyway, your comment can explain the perf difference as SharedPool requires us to copy the attributes for safe processing later, while ThreadLocalPool does not need to :)

Reply:
> SharedPool requires us to copy the attributes for safe processing later

You mean `BatchProcessor` requires us to copy the attributes for safe processing later?

Reply:
> > SharedPool requires us to copy the attributes for safe processing later
> 
> You mean `BatchProcessor` requires us to copy the attributes for safe processing later?

Yes.

Reply:
You could update the PR description as well. It mainly attributes thread pooling differences to make a case for adding a `BatchProcessor` benchmark.